RuntimeDetour/Platforms/Runtime/DetourRuntimeNETPlatform.cs

```diff
@@ -151,8 +151,42 @@ class DetourRuntimeNETPlatform : DetourRuntimeILPlatform {
             // IMPORTANT: IN SOME CIRCUMSTANCES, THIS CAN FIND ThePreStub AS THE ENTRY POINT.
 
             if (PlatformHelper.Is(Platform.ARM)) {
-                // TODO: Debug detouring NGEN'd methods on ARM.
+                if (IntPtr.Size == 4) {
+                    // TODO: 32-bit arm precode walkers
+                } else {
+                    IntPtr WalkPrecode(IntPtr curr) {
+                        long lptr = (long) curr;
+
+                        if (
+                            *(uint*) (lptr + 0x00) == 0x1000000c && // adr x12, #0x00
+                            *(uint*) (lptr + 0x04) == 0x5800006b && // ldr x11, #0x0c
+                            *(uint*) (lptr + 0x08) == 0xd61f0160    // br x11
+                        ) {
+                            IntPtr next = *(IntPtr*) (lptr + 0x10);
+                            return NotThePreStub(curr, next);
+                        } else if (
+                            *(uint*) (lptr + 0x00) == 0x10000089 && // adr x9, #0x10
+                            *(uint*) (lptr + 0x04) == 0xa940312a && // ldp x10, x12, [x9]
+                            *(uint*) (lptr + 0x08) == 0xd61f0140    // br x10
+                        ) {
```

Comment:
It looks like this is generated by https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/vm/arm64/stubs.cpp#L568. It looks like there is also a ThisPtrRetBuf stub shortly thereafter on line 590 and 641 that you're not checking for.

The first is generated at https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/vm/arm64/cgencpu.h#L688 as well. It would probably beneficial to link those in comments. We should also find the equivalents in x86 and link them, though the x86 code also performs scanning for NetFX which has some other quirks.

Reply:
Some relevant locations for x86 are:
- https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/vm/amd64/cgencpu.h#L612
- https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/vm/amd64/cgencpu.h#L733 (We only handle this one with the auto-scanning that was recently implemented)
- https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/vm/amd64/cgenamd64.cpp#L802 (This file contains quite a few stubs; I'm not sure which ones are possible in this position)
- https://github.com/dotnet/runtime/blob/4da6b9a8d55913c0ea560d63590d35dc942425be/src/coreclr/vm/amd64/virtualcallstubcpu.hpp may also generate potentially relevant code

Reply:
@0x0ade You may be interested in the above.

Reply:
I added the remaining arm64 ones -- didn't really want to look at the amd64 ones and annotate them...